[chuahziyang] ip

chuahziyang opened the pull request:
# ✨ Feature: Enhanced Todo List Management

This PR implements improved todo list functionality with better task organization capabilities.

## Key Changes
* Added task prioritization
* Implemented due date sorting
* Added task categories/tags
* Enhanced search functionality

## Implementation Steps
1. Updated `TodoItem` class
2. Added filtering system
3. Enhanced UI components

### Code Example
```typescript
interface TodoItem {
  id: string;
  title: string;
  priority: 'high' | 'medium' | 'low';
  dueDate?: Date;
  tags: string[];
  completed: boolean;
}

function sortTodosByPriority(todos: TodoItem[]): TodoItem[] {
  const priorityOrder = { high: 1, medium: 2, low: 3 };
  return todos.sort((a, b) => 
    priorityOrder[a.priority] - priorityOrder[b.priority]
  );
}
```

## Tasks
- [x] Core todo management features
- [x] Priority sorting implementation
- [ ] UI components update
- [ ] E2E testing

> This update improves task organization and helps users better manage their priorities

As Steve Jobs said:
> Simple can be harder than complex: You have to work hard to get your thinking clean to make it simple.

For implementation details, see our [component design](https://example.com/todo-design).

The new `sortTodosByPriority()` function replaces the ~~basic date-based sorting~~ with a more **intuitive** and *priority-focused* approach.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chuahziyang committed:
Level-0

chuahziyang committed:
Level-1

chuahziyang committed:
Level-2

chuahziyang committed:
Level-3

chuahziyang committed:
Level-4

chuahziyang committed:
A-TextUiTesting

chuahziyang committed:
Level-5

chuahziyang committed:
Level-6

chuahziyang committed:
Level-7

chuahziyang committed:
refac

chuahziyang committed:
Level-8

chuahziyang committed:
A-MoreOOP

chuahziyang committed:
A-Packages

chuahziyang committed:
Merge remote-tracking branch 'origin/add-gradle-support'

chuahziyang committed:
A-Gradle

chuahziyang committed:
A-JUnit

chuahziyang committed:
Level-9

chuahziyang committed:
A-JavaDoc

chuahziyang committed:
A-CodingStandard

chuahziyang committed:
Merge tag 'A-javaDoc'

chuahziyang committed:
Merge tag 'Level-9'

chuahziyang committed:
update

farhan-navas commented on src/main/java/ziyang/Ziyang.java:
```diff
@@ -0,0 +1,34 @@
+
+public class Ziyang {
+    public static void main(String[] args) throws FileNotFoundException, IOException {
+        UI ui = new UI();
```
<code>
Ui ui = new Ui();
</code>

Shouldn't the class name be Ui to follow conventions instead of UI?

farhan-navas commented on src/main/java/ziyang/Ziyang.java:
```diff
@@ -0,0 +1,34 @@
+package ziyang;
```
This file is missing JavaDoc comments

farhan-navas commented on src/main/java/ziyang/deadlineTask.java:
```diff
@@ -0,0 +1,32 @@
+
+    /**
+     * Constructs a new deadline task.
+     * @param description The description of the task
```
@param should be ended with a full stop, 

[[Link to relevant line in guide]](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Punctuation%20behind%20each%20parameter%20description.)

farhan-navas reviewed (commented):
Overall LGTM, just some coding standard issues if not very well done!

szelingtan commented on src/main/java/ziyang/Ziyang.java:
```diff
@@ -0,0 +1,34 @@
+        Parser parser = new Parser(items);
+
+        ui.welcome();
+
```
Should there be a double line spacing here?

szelingtan commented on src/main/java/ziyang/Parser.java:
```diff
@@ -0,0 +1,102 @@
+
+public class Parser {
+
+    public String nextinput;
```
Perhaps the variable can be written in camelCase

szelingtan commented on src/main/java/ziyang/Task.java:
```diff
@@ -0,0 +1,40 @@
+    protected String description;
+    
+    /** The completion status of the task */
+    protected boolean isDone;
```
I like how you made your boolean variable to sound like boolean

szelingtan reviewed (commented):
Overall, I found your code easy to read and understand. Keep it up and remember to take note of variable naming!

szelingtan commented on src/main/java/ziyang/eventTask.java:
```diff
@@ -0,0 +1,37 @@
+
+    /**
+     * Constructs a new event task with description and time period.
+     * @param description the description of the event
```
Should there be an empty line between description and parameter section?

szelingtan commented on src/main/java/ziyang/eventTask.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    public eventTask(String description, String start, String end) {
+        super(description);
+        LocalDate startdate = LocalDate.parse(start);
```
Perhaps the variable "startdate" can be in camelCase?

szelingtan commented on src/main/java/ziyang/deadlineTask.java:
```diff
@@ -0,0 +1,32 @@
+ * Represents a task with a deadline date.
+ * Extends the base Task class.
+ */
+public class deadlineTask extends Task {
```
Perhaps the class name "deadlineTask" can be written in PascalCase?

szelingtan commented on src/main/java/ziyang/Ziyang.java:
```diff
@@ -0,0 +1,34 @@
+package ziyang;
+
+import java.io.FileNotFoundException;
```
I like how you listed your imported classes explicitly

szelingtan reviewed (commented):
Added a few more comments to take note of for naming variables and classes! I also noted on a mistake in some of your Javadoc comments

Neilchen863 commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,12 @@
+
+public class UI {
```
I think the class UI should be named as Ui according to the coding style.

Neilchen863 commented on src/main/java/Ziyang.java:
```diff
@@ -8,95 +8,30 @@
+      UI ui = new UI();
+      Storage storage = new Storage("database.data");
+      
+      LinkedList<Task> items = storage.read();
```
Good naming! Using items instead of item.

Neilchen863 commented on src/test/java/ziyang/todoTaskTest.java:
```diff
@@ -0,0 +1,19 @@
+
+public class todoTaskTest {
+    @Test
+    public void dummyTest(){
```
I think the method name dummyTest needs to be updated according to the specific feature you are going to test.

Neilchen863 commented on src/main/java/Ziyang.java:
```diff
@@ -62,9 +84,14 @@ else if (input.startsWith("event")) {
           }
+          out.writeObject(items);
+          //out.close();
+          //fileOut.close();
```
What is the meaning of this comment?

Neilchen863 commented on src/main/java/ziyang/Parser.java:
```diff
@@ -57,6 +57,51 @@ public String evaluate() {
+      return msg;
+    }
+    else if (this.nextinput.startsWith("todo")) {
+      String j = this.nextinput.split(" " ,2)[1];
```
I think the String name j can be changed to like taskName, which can make your code more readable.

Neilchen863 reviewed (commented):
Overall pretty good. Cheers!

chuahziyang committed:
added checkstyle

chuahziyang committed:
part 3

chuahziyang committed:
extract to fxml

chuahziyang committed:
Level-10 done

chuahziyang committed:
finish branch-level-10 alr

chuahziyang committed:
Merge remote-tracking branch 'origin/Level-10'

chuahziyang committed:
Added shorthand for all the commands

There are several commands that can be used to mark tasks as done, delete tasks, and add new tasks. These commands can be abbreviated to save time and make the commands more concise. For example, instead of typing "mark" and "delete", you can simply type "m" and "d" respectively.

For example, if you want to mark a task as done, you can type "m 1" instead of "mark 1". Similarly, you can use "d" to delete a task.

The "list" command can be used to display the list of tasks in the database. This command can be abbreviated to "l".

The "todo" command can be used to add a new todo task. This command can be abbreviated to "t".

The "deadline" command can be used to add a new deadline task. This command can be abbreviated to "d".

The "event" command can be used to add a new event task. This command can be abbreviated to "e".

This will make it easier for the user to manage their tasks and stay organized.

chuahziyang committed:
Userguide

chuahziyang committed:
fix: fix readme file

chuahziyang committed:
feat: remove excess line

chuahziyang committed:
fix: clarify deadline and event

chuahziyang committed:
Merge pull request #2 from chuahziyang/mergebranch

fix: clarify deadline and event

chuahziyang committed:
fix: edit image path

chuahziyang committed:
fix: edit image path

chuahziyang committed:
fix: change the name of the jar file

chuahziyang committed:
fix: change the name of the jar file and add the manifest file

chuahziyang committed:
fix: change png name
